Migrated Kafka Connect Elasticsearch sink connector

ftisiot opened the pull request:
# What changed, and why it matters

Migrated Kafka Connect Elasticsearch sink connector https://help.aiven.io/en/articles/2349675-aiven-for-apache-kafka-elasticsearch-sink-connector

Also adds Kafka Connect Opensearch sink connector

Fixes #566, #568

anelook commented on docs/products/kafka/kafka-connect/howto/opensearch-sink.rst:
```diff
@@ -0,0 +1,111 @@
+
+.. Note::
+
+    If you're using Aiven for OpenSearch® and Aiven for Apache Kafka® the above details are available in the `Aiven console <https://console.aiven.io/>`_ service Overview tab or via the dedicated ``avn service get`` command with the :ref:`Aiven CLI <avn_service_get>`.
```
```suggestion
    If you're using Aiven for OpenSearch® and Aiven for Apache Kafka® the above details are available in the `Aiven console <https://console.aiven.io/>`_ service *Overview tab* or via the dedicated ``avn service get`` command with the :ref:`Aiven CLI <avn_service_get>`.
```

anelook commented on docs/products/kafka/kafka-connect/howto/opensearch-sink.rst:
```diff
@@ -0,0 +1,111 @@
+The configuration file contains the following entries:
+
+* ``name``: the connector name
+* ``OS_CONNECTION_URL``, ``OS_USERNAME``, ``OS_PASSWORD``: sink OpenSearch parameters collected in the :ref:`prerequisite <connect_opensearch_sink_prereq>` phase. 
```
I don't see these parameters in the configuration file

ftisiot replied:
Good one, solved

anelook commented on docs/products/kafka/kafka-connect/howto/opensearch-sink.rst:
```diff
@@ -0,0 +1,111 @@
+        "value.converter.schema.registry.basic.auth.user.info": "SCHEMA_REGISTRY_USER:SCHEMA_REGISTRY_PASSWORD"
+    }
+
+The configuration file contains the following entries:
```
Some of the params below don't correspond to the file above

ftisiot replied:
Which ones?

anelook commented on docs/products/kafka/kafka-connect/howto/opensearch-sink.rst:
```diff
@@ -0,0 +1,111 @@
+        "connector.class": "io.aiven.kafka.connect.opensearch.OpensearchSinkConnector",
+        "topics": "TOPIC_LIST",
+        "connection.url": "ES_CONNECTION_URL",
+        "connection.username": "ES_USERNAME",
```
should be not ES, but OS in the example

anelook commented on docs/products/kafka/kafka-connect/howto/opensearch-sink.rst:
```diff
@@ -0,0 +1,111 @@
+The configuration file contains the following entries:
+
+* ``name``: the connector name
+* ``OS_CONNECTION_URL``, ``OS_USERNAME``, ``OS_PASSWORD``: sink OpenSearch parameters collected in the :ref:`prerequisite <connect_opensearch_sink_prereq>` phase. 
```
For consistency with other params, we should use parameter names, not our variables - ``connection.username``, etc. 

Here, and in the section for Elasticsearch

ftisiot replied:
fixed

anelook commented on docs/products/kafka/kafka-connect/howto/elasticsearch-sink.rst:
```diff
@@ -0,0 +1,111 @@
+        "connection.username": "ES_USERNAME",
+        "connection.password": "ES_PASSWORD",
+        "type.name": "TYPE_NAME",
+        "tasks.max":"NR_TASKS",
```
we're never referencing NR_TASKS. maybe use "1" instead?

anelook commented on docs/products/kafka/kafka-connect/howto/opensearch-sink.rst:
```diff
@@ -0,0 +1,111 @@
+        "connection.username": "ES_USERNAME",
+        "connection.password": "ES_PASSWORD",
+        "type.name": "TYPE_NAME",
+        "tasks.max":"NR_TASKS",
```
we're never referencing NR_TASKS. maybe use "1" instead?

anelook commented on docs/products/kafka/kafka-connect/howto/opensearch-sink.rst:
```diff
@@ -0,0 +1,111 @@
+
+:: 
+
+    avn service connector create SERVICE_NAME @opensearch_sink.json
```
SERVICE_NAME here is the kafka service name or opensearch? or a new one?

anelook reviewed (changes requested):
I have a feeling we are omitting many details available in the original article, so it makes it difficult to setup the connection. Can we add some examples? also, where to find properties for kafka service APACHE_KAFKA_HOST:SCHEMA_REGISTRY_PORT, USER_INFO, SCHEMA_REGISTRY_USER:SCHEMA_REGISTRY_PASSWORD, etc..
It seems that some pieces are missing :)

ftisiot commented:
Thanks for the review @anelook, I addressed some of your concerns. 
I took some bits out of the old article but they were about something not strictly needed (like how to check if a connector is working and how to check data in OS/ES), do you think the current version is ok?

anelook commented:
@ftisiot, what should go into  "key.converter.basic.auth.credentials.source" param? :D  is it reference to one of the certificates we have?

ftisiot commented:
Should be fixed!

anelook reviewed (changes requested):
Thanks! Examples helped a lot!!! I managed to create a connector almost without issues. had some errors related to URL of the open/elastic search - so added some comments on that.

One other comment is to overall structure of this article, it seems to become so big.. I wonder if we should split it into smaller pieces by bringing examples somewhere else. It's also not the easiest read, since there are so many extra parameters schemas and schema-less approach.

Annoyingly if a mistake is made in config file, the error message is not always easy to understand. So  step-by-step end-to-end guidance is something which will be appreciated by the readers, I believe.

ftisiot commented on docs/products/kafka/kafka-connect/howto/opensearch-sink.rst:
```diff
@@ -0,0 +1,237 @@
+
+.. Warning::
+
+    The current version of the OpenSearch sink connector is not able to automatically create daily indexes in OpenSearch. Therefore you need to create the indexes with the correct name before starting the sink connector. You can create OpenSearch indexes in many ways including :doc:`CURL commands </docs/products/opensearch/howto/opensearch-with-curl>`.
```
@anelook I don't get this comment (probably too early in the morning), what is the repetition here? I'm stating:

* The connector doesn't create automatically
* You need to create them
* You can create them with CURL <- maybe we can switch to `avn service index create`?
*

ftisiot commented:
@anelook moved from CLI to Console instructions

anelook commented on docs/products/kafka/kafka-connect/howto/elasticsearch-sink.rst:
```diff
@@ -0,0 +1,243 @@
+
+To create the connector, access the `Aiven Console <https://console.aiven.io/>`_ and select the Aiven for Apache Kafka® or Aiven for Apache Kafka Connect® service where the connector needs to be defined, then:
+
+* Click on the **Connectors** tab
```
I think this should be numbered, better for steps than bullet point list

anelook commented on docs/products/kafka/kafka-connect/howto/elasticsearch-sink.rst:
```diff
@@ -0,0 +1,243 @@
+
+    Connectors can be created also using the dedicated :ref:`Aiven CLI command <avn_service_connector_create>`.
+
+Create daily Elasticsearch indexes
```
Let's use "indices" instead for plural. Since the docs use them https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-indices/ https://www.elastic.co/guide/en/elasticsearch/reference/current/cat-indices.html

anelook commented on docs/products/kafka/kafka-connect/howto/elasticsearch-sink.rst:
```diff
@@ -0,0 +1,243 @@
+
+.. Warning::
+
+    The current version of the Elasticsearch sink connector is not able to automatically create daily indexes in Elasticsearch. Therefore you need to create the indexes with the correct name before starting the sink connector. You can create Elasticsearch indexes in many ways including :doc:`CURL commands </docs/products/opensearch/howto/opensearch-with-curl>`.
```
There are 8 places in the PR with indexes - Let's use "indices" everywhere. Since the docs use them https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-indices/ https://www.elastic.co/guide/en/elasticsearch/reference/current/cat-indices.html

ftisiot replied:
fixed

anelook reviewed (changes requested):
Thx for adding the changes. I left couple of minor comments.

The pieces still look long to me, but since we didn't come to an agreement, we can leave them as they are and refactor later once we decide on the github repo approach.
